src/models/PlantModel.ts

```diff
@@ -0,0 +1,75 @@
+import { HandleFile, FileType } from './HandleFile';
+import { IPlant, IOpsInfo } from '../interfaces';
+
+class PlantModel {
+  private handleFile = new HandleFile();
+
+  private async updateOpsInfo(incrementAmount = 1): Promise<number> {
+    const opsInfo = await this.handleFile.readFile<IOpsInfo>(FileType.OpsInfo);
+    opsInfo.createdPlants += incrementAmount;
+
+    await this.handleFile.saveFile(FileType.OpsInfo, opsInfo);
+
+    return opsInfo.createdPlants;
+  }
+
+  public async getAll(): Promise<IPlant[]> {
+    const plants = await this.handleFile.readFile<IPlant[]>(FileType.Plants);
+    return plants;
+  }
+
+  public async create(plant: Omit<IPlant, 'id'>): Promise<IPlant> {
+    const plants = await this.getAll();
+
+    const newPlantId = await this.updateOpsInfo(1);
+    const newPlant = { id: newPlantId, ...plant };
+    plants.push(newPlant);
+
+    await this.handleFile.saveFile(FileType.Plants, plants);
+
+    return newPlant;
+  }
+
+  public async getById(id: string): Promise<IPlant | null> {
+    const plants = await this.getAll();
+
+    const plantById = plants.find((plant) => plant.id === parseInt(id, 10));
+    if (!plantById) return null;
+    return plantById;
+  }
+
+  public async removeById(id: string): Promise<IPlant | null> {
+    const plants = await this.getAll();
+
+    const removedPlant = plants.find((plant) => plant.id === parseInt(id, 10));
+    if (!removedPlant) return null;
+
+    const newPlants = plants.filter((plant) => plant.id !== parseInt(id, 10));
+    this.handleFile.saveFile(FileType.Plants, newPlants);
```

Comment:
```suggestion
    this.handleFile.saveFile(FileType.Plants, newPlants);
    
    const REMOVE_ONE_PLANT = -1;
    await this.updateOpsInfo(REMOVE_ONE_PLANT); 
```
O numerador em opsInfo.json não estava sendo alterado quando uma planta era excluída.

Reply:
Isto é proposital, uma vez que o opsInfo.json te, a função de armazenar a `PRIMARY KEY` com `AUTO_INCREMENT`.